### PR TITLE
Module storage updates

### PR DESCRIPTION
**Pull request by And1sS**

### CODE REVIEW

File: src/main/java/org/prebid/cache/config/StorageConfig.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.annotation.Validated;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Configuration
```
Comment (AntoxaAntoxic) [inline src/main/java/org/prebid/cache/config/StorageConfig.java:13]: shouldn't it work without `@Configuration`?

File: src/main/java/org/prebid/cache/config/StorageConfig.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@Validated
```
Comment (AntoxaAntoxic) [inline src/main/java/org/prebid/cache/config/StorageConfig.java:14]: this one is not utilized, or some validation annotations are missing

File: src/main/java/org/prebid/cache/handlers/storage/PostStorageHandler.java [hunk 3]
```diff
@@ -66,11 +69,15 @@ private void validateModulePayload(final ModulePayload payload, final Synchronou
         }
     }
 
-    private static PayloadWrapper mapToPayloadWrapper(final ModulePayload payload) {
+    private PayloadWrapper mapToPayloadWrapper(final StoragePayload payload) {
+        final long ttlSeconds = Optional.ofNullable(payload.getTtlseconds())
+                .map(Integer::longValue)
+                .orElse(storageConfig.getDefaultTtlSeconds());
```
Comment (AntoxaAntoxic) [inline src/main/java/org/prebid/cache/handlers/storage/PostStorageHandler.java:75]: not covered by unit tests

**Review by AntoxaAntoxic — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/cache/config/StorageConfig.java (hunk 1 above)
Comment (And1sS) [reply src/main/java/org/prebid/cache/config/StorageConfig.java:15]: No, for that to work without `@Configuration` we need to change component scan method.

File: src/main/java/org/prebid/cache/config/StorageConfig.java (hunk 2 above)
Comment (And1sS) [reply src/main/java/org/prebid/cache/config/StorageConfig.java:16]: Nice catch, changed that!

File: src/main/java/org/prebid/cache/handlers/storage/PostStorageHandler.java (hunk 3 above)
Comment (And1sS) [reply src/main/java/org/prebid/cache/handlers/storage/PostStorageHandler.java:75]: Added unit test.

**Review by AntoxaAntoxic — DISMISSED** [review]

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**